### ISSUE TITLE
Hot restart/reload broken after flutter upgrade and flutter downgrade on Android Studio

### ISSUE THREAD

**Issue by vinnytwice** [flutter/flutter#72671]
Yesterday I did run flutter upgrade but then the `intl` package was giving some version incompatibility issues so after a few tries I just run flutter downgrade to go back to previous version.
Now my app goes crazy when I hot restart it.. it works normally if I uninstall it an
 reinstall it again, bau as soon as I press hot restart or hot reload it goes crazy like in a loop trying to rebuild the view, so I have to uninstall it again..

Have you experienced this situation? 
What can I try before reinstalling Android Studio?

This is my flutter doctor.
Thank you very much.
Cheers.

 ```
[✓] Flutter (Channel master, 1.23.0-19.0.pre.20, on Mac OS X 10.15.4 19E287 darwin-x64, locale en-US)
    • Flutter version 1.23.0-19.0.pre.20 at /Users/vinnytwice/Developer/flutter
    • Framework revision 66cf8d47a3 (10 weeks ago), 2020-10-14 14:47:28 +0900
    • Engine revision adf5b59485
    • Dart version 2.11.0 (build 2.11.0-217.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Volumes/ProjectsSSD/Android/sdk
    • Platform android-29, build-tools 30.0.2
    • ANDROID_HOME = /Volumes/ProjectsSSD/Android/sdk
    • ANDROID_SDK_ROOT = /Volumes/ProjectsSSD/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Volumes/ProjectsSSD/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (3 available)
    • iPad (7th generation) (mobile) • CF798CBE-3FBF-47E3-A7DF-8E762B4B8CEC • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
    • Web Server (web)               • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)                   • chrome                               • web-javascript • Google Chrome
      87.0.4280.88

• No issues found!

```

**Comment by iapicca** [comment]
@vinnytwice 
can you run `flutter upgrade -f`
if the issue persists please provide
your update `flutter doctor -v`
and your `flutter run -v `from terminal using both `r` and `R`

**Comment by pedromassangocode** [comment]
Hi @vinnytwice 
You are using a old version of Flutter, please see https://github.com/flutter/flutter/issues/72671#issuecomment-748584358 and provide what was requested there.

**Comment by Macacoazul01** [comment]
i'm with the same problem. Everytime i press the hot restart (or reload) button, visual code gets stuck in an eternal loop and nothing happens. This issue is present in the auto reload and in the manual reload.

So, i'm having to close and restart the web app everytime i want to see the changes.

![image](https://user-images.githubusercontent.com/57464965/103152423-8e9fb680-4766-11eb-817d-a18926787dea.png)

![image](https://user-images.githubusercontent.com/57464965/103152454-f2c27a80-4766-11eb-904d-03a9f4c54590.png)

my flutter doctor:

Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 1.26.0-2.0.pre.137, on Microsoft Windows [versÃ£o 10.0.19042.685], locale pt-BR)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.52.1)
[√] Connected device (2 available)

• No issues found!
PS C:\Users\gian_\Desktop\bianca_web_app> flutter doctor -v
[√] Flutter (Channel master, 1.26.0-2.0.pre.137, on Microsoft Windows [versÃ£o 10.0.19042.685], locale pt-BR)
    • Flutter version 1.26.0-2.0.pre.137 at C:\flutter
    • Framework revision 8f5d0371af (6 hours ago), 2020-12-26 11:07:36 +0330
    • Engine revision f5364860ab
    • Dart version 2.12.0 (build 2.12.0-179.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\gian_\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.52.1)
    • VS Code at C:\Users\gian_\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.0

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 87.0.664.60

• No issues found!

**Comment by vinnytwice** [comment]
@iapicca Hi Francesco, this is my flutter doctor before upgrading:
```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E287 darwin-x64, locale en-US)
    • Flutter version 1.22.5 at /Users/vinnytwice/Developer/flutter
    • Framework revision 7891006299 (3 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Volumes/ProjectsSSD/Android/sdk
    • Platform android-29, build-tools 30.0.2
    • ANDROID_HOME = /Volumes/ProjectsSSD/Android/sdk
    • ANDROID_SDK_ROOT = /Volumes/ProjectsSSD/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Volumes/ProjectsSSD/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.10.0

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (1 available)
    • iPad (7th generation) (mobile) • CF798CBE-3FBF-47E3-A7DF-8E762B4B8CEC • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)

• No issues found!

```

running `flutter upgrade -f` got me this:
``` upgrade -f
Flutter is already up to date on channel stable
Flutter 1.22.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 7891006299 (3 weeks ago) • 2020-12-10 11:54:40 -0800
Engine • revision ae90085a84
Tools • Dart 2.10.4

**Comment by vinnytwice** [comment]
[Logs.zip](https://github.com/flutter/flutter/files/5757463/Logs.zip)


As the logs are too long for a comment here I zipped a .txt to upload them here.
Thanks

**Comment by pedromassangocode** [comment]
Hi @vinnytwice 
Based on the logs it seems that your project has issues unrelated to hot-reload. Are you are to reproduce the issue on a new Flutter project?

**Comment by pedromassangocode** [comment]
Hi @vinnytwice 
Thanks for the info, without any valid reproducible code sample I cannot consider this as a valid issue. I would suggest you to try to find out a way to reproduce the issue without exposing your private files.
Thank you

**Comment by Macacoazul01** [comment]
@vinnytwice i've the same problem as you. Had to create a new project to use the hot-restart function again

**Comment by vinnytwice** [comment]
@Macacoazul01 what did you do? I have crated a new project and imported/swapped or copied/pasted relevant files but then the issue started again.. 
I might try again if you did something different

**Comment by Macacoazul01** [comment]
@vinnytwice 
I've also added an exception for the flutter folder on my antivirus .
Try not importing all the project files for the new one at the same time, to find if there is one that causes the problem

**Comment by vinnytwice** [comment]
@pedromassangocode Right, I'll try it and get back with results. Yesterday dough another project ( same apps bundle ) started doing the same.. so I might have to dig deeper into the problem..

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by bambinoua** [comment]
> 
> 
> @pedromassangocode well actually if I create a new project then it does hot-restart normally, but if then copy files from the failing project(restaurant app) to the newly created project then it fails too.
I confim this. After upgrade flutter one of my web project cannot be hot restarted. Other simple project is working.

**Comment by bambinoua** [comment]
I have some additional information about this issue.
![hot_restart_error](https://user-images.githubusercontent.com/45417992/119157317-bd902f00-ba5d-11eb-8aa7-25c75af0341f.jpg)

**Comment by bambinoua** [comment]
Exception details:

![hot_restart_error_2](https://user-images.githubusercontent.com/45417992/119157968-62127100-ba5e-11eb-9ca8-aa260fc4ec93.jpg)

**Comment by bambinoua** [comment]
I used MaterialApp.router with some package. I tried to use default MaterialApp widget with just some `home` widget. When I changed home widget content and click Save I save in VS Code `Restarted application in 490ms`, in Chrome console I saw `3 module(s) were hot-reloaded` but the page was not refreshed (just a blank page). Whan I refresh browser in console I saw `Loading from service worker` and the the page is refreshed.
